tests/buffer.spec.ts

```diff
@@ -80,14 +80,24 @@ describe("buffer", () => {
     expect<i32>(buff.writeUInt8(252,4)).toBe(5);
     expect<u8>(buff[0]).toBe(4);
     expect<u8>(buff[4]).toBe(252);
-  });
+    // TODO:
+    // expectFn(() => { 
+    //   let newBuff = new Buffer(1);
+    //   newBuff.writeUInt8(5,10);
+    // }).toThrow();    
+  });  
 
   test("#writeInt8", () => {
     let buff = new Buffer(5);
     expect<i32>(buff.writeInt8(9)).toBe(1);
     expect<i32>(buff.writeInt8(-3,4)).toBe(5);
     expect<i8>(buff[0]).toBe(9);
```

Comment:
Each one of these `write` functions do granular tests of each byte.

This isn't really all that nice looking. Can you please use:

```ts
function create<T>(values: valueof<T>[]): T {
  let result = instantiate<T>(values.length);
  for (let i = 0; i < values.length; i++) result[i] = values[i];
  return result;
}

  test("#writeInt8", () => {
    let buff = new Buffer(5);
    expect<i32>(buff.writeInt8(9)).toBe(1);
    expect<i32>(buff.writeInt8(-3,4)).toBe(5);
    let expected = create<Buffer>([0x09, 0x0, 0x0, 0x0, 0xFD]);
    expect<Buffer>(buff).toStrictEqual(expected);
  });
```

Sorry for the large nitpick. but this will be much easier to maintain and easier to understand for others.

Reply:
Implemented that check.